api/catalog/api/models/audio.py

```diff
@@ -209,3 +223,26 @@ class Meta:
     def save(self, *args, **kwargs):
         self.slug = uuslug(self.title, instance=self)
         super(AudioList, self).save(*args, **kwargs)
+
+
+class AudioWaveformAddOn(AbstractMediaAddOn):
+    audio = models.OneToOneField(
+        to=Audio,
+        on_delete=models.CASCADE,
+        related_name="waveform",
+        help_text=("The foreign key of the audio."),
+    )
+
+    peaks = ArrayField(
```

Comment:
Instead of the strict typing as an array of floats, we could possibly use a `jsonb` field? That allows us to tweak waveform date if needed in the future & prevents the need for an approximation of 1500.

Reply:
I thought about using `jsonb` but I actually prefer the strict typing. I wonder if there are any performance characteristics we should consider too, if inserting an array is faster than jsonb over time? For a single client it's possible that up to 10 waveform requests will be made for an audio search result page which could cause a pretty large number of transactions under heavy traffic.

Reply:
To boost the performance, I was at some point considering compressing the JSON data of the waveform and storing it as a text field in the DB 😄. I'm not sure which of our choices is most performant, maybe @AetherUnbound has more knowledge in that area, but imo, we should aim for max performance even if the cost is absence of validation.